lsp-types/src/Language/LSP/VFS.hs

```diff
@@ -341,9 +333,8 @@ getCompletionPrefix pos@(J.Position l c) (VirtualFile _ _ ropetext) =
             lastMaybe [] = Nothing
             lastMaybe xs = Just $ last xs
 
-        curLine <- headMaybe $ T.lines $ Rope.toText
-                             $ fst $ Rope.splitAtLine 1 $ snd $ Rope.splitAtLine (fromIntegral l) ropetext
-        let beforePos = T.take (fromIntegral c) curLine
+        let curRope = fst $ Rope.splitAtLine 1 $ snd $ Rope.splitAtLine (fromIntegral l) ropetext
+        beforePos <- Rope.toText . fst <$> Rope.splitAt (fromIntegral c) curRope
```

Comment:
BTW this fixes an existing bug: `T.take` counts code points, while LSP requires counting UTF-16 code units.